apollo-compiler/src/test/graphql/com/example/arguments_complex/TestQuery.kt

```diff
@@ -126,19 +126,21 @@ data class TestQuery(
      */
     val height: Double?
   ) {
-    fun marshaller(): ResponseFieldMarshaller = ResponseFieldMarshaller.invoke { writer ->
-      writer.writeString(RESPONSE_FIELDS[0], this@HeroWithReview.__typename)
-      writer.writeString(RESPONSE_FIELDS[1], this@HeroWithReview.name)
-      writer.writeDouble(RESPONSE_FIELDS[2], this@HeroWithReview.height)
+    fun marshaller(): ResponseFieldMarshaller {
+      return ResponseFieldMarshaller.invoke { writer ->
+        writer.writeString(RESPONSE_FIELDS[0], this@HeroWithReview.__typename)
+        writer.writeString(RESPONSE_FIELDS[1], this@HeroWithReview.name)
+        writer.writeDouble(RESPONSE_FIELDS[2], this@HeroWithReview.height)
```

Comment:
Not strictly related to this PR but the `ResponseFieldMarshaller` would be a good candidate to try out the shiny new `fun interface` in Kotlin 1.4

Reply:
yeah for sure, we can tune it later